packages/solana/src/utils/wallet-standard/watchStandard.ts

```diff
@@ -0,0 +1,35 @@
+import { getWallets } from '@wallet-standard/app'
+import { StandardWalletAdapter } from './adapter'
+import { isWalletAdapterCompatibleStandardWallet } from '@solana/wallet-adapter-base'
+import type { Wallet } from '@wallet-standard/base'
+
+const { get, on } = getWallets()
+
+let standardAdapters: StandardWalletAdapter[] = [...wrapWalletsWithAdapters(get())]
+
+export function watchStandard(callback: (arg: StandardWalletAdapter[]) => void) {
+  const listeners = [
+    on('register', (...wallets) => {
+      if (!standardAdapters || standardAdapters.length === 0) {
+        standardAdapters = [...wrapWalletsWithAdapters(wallets)]
+      } else {
+        standardAdapters = [...standardAdapters, ...wrapWalletsWithAdapters(wallets)]
+      }
+      callback(standardAdapters)
+    }),
+    on('unregister', (...wallets) => {
+      standardAdapters = standardAdapters.filter(standardAdapter =>
+        wallets.some(wallet => wallet === standardAdapter.wallet)
```

Comment:
Is the `.wallet` field string here? It's not understandable name and sounds like it's an object. If it's an object, this equality condition wouldn't work.

Reply:
> If it's an object, this equality condition wouldn't work.

Why not?

I took this from Solana Labs (Anza) source code btw.

Reply:
> Why not?
Because it's JavaScript. In JS worlds two objects are never the same even though they have exact same fields. And the equality condition `===` would never work with objects. They only work with primitive values (strings, boolean, number) Generally if you need such a condition, some special deep equality checker methods would have been used. See https://www.syncfusion.com/blogs/post/deep-compare-javascript-objects

If Solana Labs using that way, prob wallet is not an object, otherwise they are doing wrong too.